webapp/src/containers/createIncident/index.tsx

```diff
@@ -78,7 +78,9 @@ const CreateIncident = () => {
     // Hide the modal and reset the states
     const hideModal = useCallback(() => {
         dispatch(resetCurrentModalState());
-        resetFieldStates();
+        setTimeout(() => {
```

Comment:
You should also use `clearTimeout` on `componentWillUnmount` wherever you're using `setTimeout`.

Reply:
discussed, it will be done in the next PR